lib/Onyx.ts

```diff
@@ -484,11 +484,13 @@ function mergeCollection<TKey extends CollectionKeyBase, TMap>(collectionKey: TK
                 return obj;
             }, {}) as Record<OnyxKey, OnyxInput<TKey>>;
 
-            const newCollection = newKeys.reduce((obj: OnyxInputKeyValueMapping, key) => {
-                // eslint-disable-next-line no-param-reassign
-                obj[key] = mergedCollection[key];
-                return obj;
-            }, {}) as Record<OnyxKey, OnyxInput<TKey>>;
+            const newCollection: Record<OnyxKey, OnyxInput<TKey>> = {};
+            keys.forEach((key) => {
+                if (persistedKeys.has(key)) {
+                    return;
+                }
+                newCollection[key] = mergedCollection[key];
+            });
```

Comment:
Looks good to me, but just curios... is the reason we use `forEach` instead of `reduce` because it's faster or more readable?

Reply:
@chrispader Only reason here is to avoid 2 loops. First we do a filter to get the `newKeys` and then we do a reduce over them. However, we could do this in one loop as `keys` is a `Set` which gives us a `forEach` helper function.